sre_yield/__init__.py

```diff
@@ -216,16 +216,13 @@ def get_item(self, i, d=None):
         if i < 0 or i >= self.length:
             raise IndexError("Index %d out of bounds" % (i,))
 
-        if len(self.list_lengths) == 1:
-            # skip unnecessary ''.join -- big speedup
-            return self.list_lengths[0][0][i]
```

Comment:
This could be restored by also checking if `d` is false-ish.
Then the perf hit only occurs for AllMatches, and not AllStrings.

Even then we can avoid the divmod by replacing here with (roughly)
```py
c = self.list_lengths[0][0]
if hasattr(c, "get_item"):
    return c.get_item(i, d)
else:
    return c[i]
```